File: src/main/java/com/clearcont/clearcontapp/repository/EmpresaRepository.java
package com.clearcont.clearcontapp.repository;

import com.clearcont.clearcontapp.model.Empresa;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.Optional;

@Repository
public interface EmpresaRepository extends JpaRepository<Empresa, Long> {
    Optional<Empresa> findEmpresaByNomeEmpresa(String nome);
}

File: src/main/kotlin/br/com/clearcont/clearcontwebapp/repository/EmpresaRepository.kt
package br.com.clearcont.clearcontwebapp.repository

import br.com.clearcont.clearcontwebapp.models.Empresa
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import java.util.*

@Repository
interface EmpresaRepository : JpaRepository<Empresa, Long> {
    fun findEmpresaByNomeEmpresa(nome: String): Optional<Empresa>
}
